aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
+
+if_statement: 'if' expression NEWLINE body
+
+switch_statement: 'switch' identifier cases
+
+case: 'case' expression [body] ['next']
+
+
+identifier: NAME
+          | identifier '.' identifier
+
+identifier_list: epsilon
+               | identifier
+               | identifier ',' identifier_list
+
+
+on: 'on' NAME ['be' NAME] functions
```

Comment:
This is the syntax for a *behavior*. Should the rule be renamed as such, or would it make more sense to keep it as `on` to match the keyword?